### ISSUE TITLE
remote kubectl not working: invalid character '\x00' in string literal

### ISSUE THREAD

**Issue by danielschlegel** [rancher/rancher#6966]
I have an nginx in front of rancher configured like this:
http://docs.rancher.com/rancher/v1.2/en/installing-rancher/installing-server/basic-ssl-config/

Rancher is accessible at https://rancher.example.com:8080
kubectl config is:
```
apiVersion: v1
kind: Config
clusters:
- cluster:
    api-version: v1
    server: "https://rancher.example.com:8080/r/projects/1a7/kubernetes"
  name: "staging"
contexts:
- context:
    cluster: "staging"
    user: "staging"
  name: "staging"
current-context: "staging"
users:
- name: "staging"
  user:
    username: "O03B679FCA5D02B96F89"
    password: "PIq1mK77ybKTKuptM1dvG9FefFP2mcaoihisrbQ6"
```
I wand to add a deployment thats pretty big.

```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: develop-gis-api-api
  namespace: develop
  labels:
    app: develop-gis-api-api
spec:
  replicas: 1
  selector:
    matchLabels:
      app: develop-gis-api-api
  template:
    metadata:
      labels:
        app: develop-gis-api-api
    spec:
      volumes:
      - name:share
        hostPath:
          path: /home/user/data/develop-gis-api-api/share
      - name: s3-share
        hostPath:
          path: /home/user/data/develop-gis-api-api/s3-share
      nodeSelector:
        kubernetes.io/hostname: hostname
      containers:
      - name: mapserver
        image: registry.example.com/mapserver:develop
        command: ['apachectl', '-DFOREGROUND']
        imagePullPolicy: Always
        ports:
          - name: mapserversta
            containerPort: 80
            protocol: TCP
        volumeMounts:
        - name: shapefiles-share
          mountPath: /app/tmp/uploads/shapefiles
        - name: s3-share
          mountPath: /mapserver/s3
      - name: api-worker
        image: registry.example.com/ac/gis-api:develop
        command: ['bash', '-c', '/entrypoint-worker.sh']
        workingDir: /app
        env:
        - name: ROLLBAR_API_KEY
          value: 'asdfasdfdasfasfddsf'
        - name: DATABASE_URL
          value: 'postgres://staging:asdfadsfdsafdsafdsafasdf@api-postgres:5432/example-api-staging'
        - name: REDIS_URL
          value: 'redis://h:asdfadsfasdf@api-redis.staging:6379'
        - name: RAILS_ENV
          value: 'production'
        - name: RACK_ENV
          value: 'production'
        - name: PORT
          value: '3000'
        - name: WEB_CONCURRENCY
          value: '5'
        - name: CORE_BASE_URL
          value: 'https://staging.example.com/'
        - name: S3_PATH
          value: '/s3'
        - name: S3_BUCKET_NAME
          value: 'develop'
        - name: S3_KEY_ID
          value: 'asdfadsf'
        - name: S3_REGION
          value: 'eu-central-1'
        - name: S3_SECRET_KEY
          value: 'sec'
        - name: SIDEKIQ_USERNAME
          value: 'agri'
        - name: SIDEKIQ_PASSWORD
          value: 'asdfasdfdsaffdas'
        - name: HTTP_USER
          value: 'agri'
        - name: HTTP_PASSWORD
          value: 'adsafasd;/asdasdfdsaf'
        - name: SECRET_KEY_BASE
          value: 'secret'
        - name: S3_USER_GROUP
          value: 'www-data:www-data'
        - name: LANG
          value: 'en_US.UTF-8'
        - name: NEW_RELIC_LICENSE_KEY
          value: 'asdfasdfdsafdasfadsfdsaf'
        - name: NEW_RELIC_LOG
          value: 'stdout'
        - name: PAPERTRAIL_API_TOKEN
          value: 'asdfasdfasdf'
        - name: SECRET_KEY_BASE
          value: 'adsfadsfdsafdsafasfdasfasdfdasf'
        - name: WEB_CONCURRENCY
          value: '1'
        - name: MAPQUEST_API_KEY
          value: 'adfsasdffdasafs'
        - name: PG_DB
          value: 'api-staging'
        - name: PG_USER
          value: 'staging'
        - name: PG_PASS
          value: 'pass'
        - name: PG_HOST
          value: 'api-postgres'
        - name: PG_PORT
          value: '5432'
        imagePullPolicy: Always
        volumeMounts:
        - name: share
          mountPath: /share
        - name: s3-share
          mountPath: /s3
```
Im able to create this deployment. 
```
kubectl -f /home/cloud/config/develop/develop-gis-api-api.deploy.yml create
deployment "develop-gis-api-api" created
```
But deletion does not work.
```
kubectl delete deployment develop-gis-api-api --namespace=develop
Error from server: the object provided is unrecognized (must be of type Deployment): couldn't get version/kind; json parse error: invalid character '\x00' in string literal ({"kind":"Deployment","apiVersi ...)
```

If i do the same command in webinterface it works e.g. https://rancher.example.com:8080/env/1a7/kubernetes/kubectl

---
| Useful | Info |
| :-- | :-- |
|Versions|Rancher `v1.2.0` Cattle: `v0.174.3` UI: `v1.2.34` |
|Access|`github` `admin`|
|Orchestration|`Kubernetes`|
|Route|`k8s-tab.kubectl`|

**Comment by danielschlegel** [comment]
https://github.com/rancher/rancher/issues/6447

**Comment by danielschlegel** [comment]
Without Nginx in front of Rancher it workes like it should but i really need a proxy in front of it. I could not find a solution for a working Nginx config now.

**Comment by danielschlegel** [comment]
Any news on this? I have several setups with the same problem.

**Comment by aemneina** [comment]
@danielschlegel it looks targeted for a fix in 1.5.0. I plan to close this issue out and we can track progress there, does that sound fair?

**Comment by danielschlegel** [comment]
Yes sure, thanks a lot for building a great product.
